client.go

```diff
@@ -143,27 +178,34 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 
 		// Attempt the request
 		resp, err := c.HTTPClient.Do(req.Request)
+
+		// check if we should continue with retries
+		checkErr := c.CheckRetry(resp, err)
+
 		if err != nil {
 			c.Logger.Printf("[ERR] %s %s request failed: %v", req.Method, req.URL, err)
-			goto RETRY
+		} else {
+			// call this here to maintain the behavior of logging all requests,
```

Comment:
This will get skipped on all non-nil errors right? including the `ErrStopRetry`. I think we need to move this out.